### PR TITLE
Upgrade debug-logtron & cleanup logs

### PR DESCRIPTION
**Pull request by Raynos**
- DebugLogtron() has enabled to to true so writes
    info, warn, error & fatal.
 - Fix log spew for socket closes
 - Fix warns around self response race condition
 - Add allocCluster.test.only() for sanity.

r: @jcorbin @kriskowal

### CODE REVIEW

File: node/test/lib/alloc-cluster.js [hunk 1]
```diff
@@ -158,6 +145,26 @@ allocCluster.test = function testCluster(desc, opts, t) {
...
+    test.only(desc, function t2(assert) {
+        allocCluster(opts).ready(function clusterReady(cluster) {
+            assert.once('end', function testEnded() {
+                cluster.destroy();
+            });
+            t(cluster, assert);
+        });
+    });
+};
+
```
Comment (kriskowal) [inline node/test/lib/alloc-cluster.js]: Consider a less copypasta solution. Perhaps thread test.only as an arg into test or a shared impl.

**Comment by kriskowal** [comment]
:ok:

File: node/test/lib/alloc-cluster.js (hunk 1 above)
Comment (Raynos) [inline node/test/lib/alloc-cluster.js]: Will create shared impl in the future.

File: node/connection_base.js [hunk 2]
```diff
@@ -307,8 +309,11 @@ TChannelConnectionBase.prototype.buildResponse = function buildResponse(req, opt
 TChannelConnectionBase.prototype.onReqDone = function onReqDone(req) {
     var self = this;
     if (self.requests.in[req.id] !== req) {
+        var inreq = self.requests.in[req.id];
+
         self.logger.warn('mismatched onReqDone callback', {
             hostPort: self.channel.hostPort,
+            inreq: !!inreq,
```
Comment (jcorbin) [inline node/connection_base.js]: how about `hasInReq: self.requests.in[req.id] !== undefined`

File: node/self_out_response.js [hunk 3]
```diff
@@ -85,8 +85,12 @@ function passError(codeString, message) {
         originalId: self.id,
         message: message
     });
-    self.inreq.outreq.errorEvent.emit(self, err);
     if (!self.closing) self.conn.lastTimeoutTime = 0;
+    process.nextTick(emitError);
+
+    function emitError() {
```
Comment (jcorbin) [inline node/self_out_response.js]: why defer

Comment (Raynos) [inline node/self_out_response.js]: necessary to avoid a race condition; can explain in person.

Comment (Raynos) [inline node/self_out_response.js]: defer should not happen here; its the wrong place; let me defer in the correct place.

**Comment by jcorbin** [comment]
Funny, I had taken a run at the info/warn log branch during xlang testing, I propose #512

File: node/self_out_response.js (hunk 3 above)
Comment (Raynos) [inline node/self_out_response.js]: no wait; this is the correct location.

**Comment by jcorbin** [comment]
lgtm, after fix inreq feedback

**Comment by Raynos** [comment]
Fixed the inreq.

**Pull request merged**